### ISSUE TITLE
eBPF core should expose batched invoke APIs

### ISSUE THREAD

**Issue by Alan-Jowett** [microsoft/ebpf-for-windows#2061]
https://github.com/microsoft/xdp-for-windows invokes an eBPF program in a tight loop for each packet.

There is overhead involved in invoking an eBPF program (acquiring/release rundown protection on the program information provider, joining/leaving the epoch).

It would make more sense to expose an advanced dispatch API that has the following functions:
1) _ebpf_link_instance_invoke_batch_begin
2) _ebpf_link_instance_invoke_batch
3) _ebpf_link_instance_invoke_end

Extension would then have the option of either calling the non-batching API, paying the cost overhead and invoking the BPF program once or calling the _batch_begin, then calling the BPF program for multiple times and finally calling the _batch_end API.

**Comment by Alan-Jowett** [comment]
@mtfriesen do you think this will resolve your concerns around eBPF invocation overhead?

**Comment by mtfriesen** [comment]
Yes, this is exactly what I was thinking. I would imagine this three-part dispatch API would be an additional option, not a replacement for the current all-in-one invoke_program dispatch routine?
